Feat: Content > Syndication

benstov opened the pull request:
### PR information
**What kind of change does this PR introduce?** (check one with "x")
- [] Bugfix
- [x] Feature
- [] Code style update (formatting, local variables)
- [] Refactoring (no functional changes, no api changes)
- [] Build related changes
- [] CI related changes
- [] Other... Please describe:

Instructions screens & iTunes destination form haven't been implemented in this branch (waiting for Product requirements)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kaltura/kmc-ng/363)
<!-- Reviewable:end -->

benstov commented:
Review status: 0 of 49 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ABRHXRKoweQY2um2T:-L4ABRHXRKoweQY2um2U:b-uzbfpa) (no related file):*
@amirch1  After push to master we need to clear the cache inside players-store.service with app-event if the players list changes 

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ABdrj9-pWfat_2oM7:-L4ABdrj9-pWfat_2oM8:b-yel0wj) (no related file):*
@esakal Need to add caching mechanism to the flavours-store.service

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 49 of 49 files at r1.
Review status: 46 of 49 files reviewed at latest revision, 36 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ABdrj9-pWfat_2oM7:-L4AIeaVxf3jTNUUTd18:buq5whq) (no related file):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

@esakal Need to add caching mechanism to the flavours-store.service
</blockquote></details>

since you did it nicely in players and also need to implement caching mechanism there - can you do it here as well?

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AE6i0v6UsgxN28usM:-L4AE6i0v6UsgxN28usN:bk44xez) (no related file):*
should add local search for players and flavors

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AEGqUBuNWyBiyU9Bf:-L4AEGqUBuNWyBiyU9Bg:b-arg8nb) (no related file):*
when creating a new feed and selecting destination there is a flicker of the validation on the destination component

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ATLuochlsHXMx6YV_:-L4ATLuochlsHXMx6YVa:beudy21) (no related file):*
should clear cache of players when added/deleted player in 'studio'

Use the event provided by Amir
https://github.com/kaltura/kmc-ng/blob/master/src/shared/content-shared/categories/categories-search.service.ts#L55

---

*[src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.html, line 1 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A-cEGe7frDvKMcVSs:-L4A-cEGe7frDvKMcVSt:bltrofl) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.html#L1)):*
> ```HTML
> <div class="kCopyToClipboardContainer">
> ```

this component should be in mc-shared

---

*[src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.html, line 19 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A0-0W9u0xQWPep_4j:-L4A0-0W9u0xQWPep_4k:bfi32ur) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.html#L19)):*
> ```HTML
>           d="M9.53 16.166l-3.853-4.128c-0.452-0.485-1.212-0.511-1.696-0.058s-0.511 1.212-0.058 1.696l4.8 5.143c0.499 0.535 1.357 0.502 1.814-0.069l9.6-12c0.414-0.518 0.33-1.273-0.187-1.687s-1.273-0.33-1.687 0.187l-8.733 10.916z"></path>
>     </svg>
>     <svg class="kError" *ngIf="_status === 'failure' || _status === 'notSupported'" version="1.1"
> ```

 'notSupported' should have a special handling. probably the same as 'idle' but disabled with the special tooltip for that state

---

*[src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.ts, line 26 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A0nRX_b5ux2fZ5Fcx:-L4A0nRX_b5ux2fZ5Fcy:brs9fv0) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.ts#L26)):*
> ```XML
>   set text(text: string) {
>     this._status = 'idle';
>     if (!this._isCopyToClipboardSupportedByBrowser() || !text || !text.length) {
> ```

not having a text doesn't equal to feature not supported. should show 'idle' icon in disable mode 

---

*[src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.ts, line 27 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A14d9dPQy1DceakVP:-L4A14d9dPQy1DceakVQ:bivlf9p) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.ts#L27)):*
> ```XML
>     this._status = 'idle';
>     if (!this._isCopyToClipboardSupportedByBrowser() || !text || !text.length) {
>       console.warn('Unable to initiate CopyToClipboardComponent, unsupported in current browser');
> ```

I think we can remove this console warning it might confuse debugging process

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 93 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A2V3qldAGT7kqKOn5:-L4A2V3qldAGT7kqKOn6:b-e79pq5) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L93)):*
> ```XML
>   private _prepare(): void {
>     if (!this._isReady) {
>       const defaultPageSize = this._browserService.getFromLocalStorage(this._pageSizeCacheKey);
> ```

in latest development we moved that part into _createDefaultFilterValue method

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 204 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A2v5NMYAozYYJGtAc:-L4A2v5NMYAozYYJGtAd:bu08r64) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L204)):*
> ```XML
>           if (feed instanceof KalturaBaseSyndicationFeed) {
>             if (feed instanceof KalturaGenericSyndicationFeed && !(feed instanceof KalturaGenericXsltSyndicationFeed)) {
>               return undefined; // stop processing this iteration if it's a generic syndication feed with XSLT type which is not generic
> ```

if you map some items to undefined you need to also filter them out by adding after the `.map` operation `.filter(Boolean)` - see http://www.devign.me/javascript-tip-remove-falsy-items-out-of-an-array

here I would add a _logger.log message indicating that feed with 'id' was removed from list since it is not....

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 220 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A3bTl-eNA3VrXcft6:-L4A3bTl-eNA3VrXcft7:buidjyn) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L220)):*
> ```XML
>   public getPlaylists(): Observable<KalturaPlaylist[]> {
>     const filter = new KalturaPlaylistFilter({orderBy: KalturaPlaylistOrderBy.createdAtDesc.toString()});
>     const pager = new KalturaFilterPager({pageSize: 500});
> ```

is it ok querying for 500 items? was it taken from the legacy?

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 284 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A46Fvu_l0N1SBGtUD:-L4A46Fvu_l0N1SBGtUE:bx63iv7) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L284)):*
> ```XML
>       .map(responses => {
>         const mergedResponses = [].concat.apply([], responses);
>         const hasFailure = mergedResponses.filter(function (response) {
> ```

since you only want to know if we have an error, `.find()` is more performant 

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 288 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4GVlvYEpaTDH5xgO:-L4A4GVlvYEpaTDH5xgP:bl1x8b3) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L288)):*
> ```XML
>         }).length > 0;
>         if (hasFailure) {
>           throw new Error('error');
> ```

please provide more concrete error message

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 296 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4IyDzvT-VwPPXABu:-L4A4IyDzvT-VwPPXABv:b-lrp1b4) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L296)):*
> ```XML
> 
> 
>   // public deleteFeed(feedId: number): Observable<void> {
> ```

cleanup?

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 392 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4W0hEVn5Xbj-UP4R:-L4A4W0hEVn5Xbj-UP4S:b-f9z3l5) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L392)):*
> ```XML
>   }
> 
>   public update(id: string, syndicationFeed: KalturaBaseSyndicationFeed): Observable<void> {
> ```

doesn't `syndicationFeed` include its' `id` value as a property? if so simplify the API

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 401 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4bYG3dizYlKLUn3S:-L4A4bYG3dizYlKLUn3T:bbouo3g) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L401)):*
> ```XML
>   public create(syndicationFeed: KalturaBaseSyndicationFeed): Observable<KalturaBaseSyndicationFeed> {
>     return this._kalturaClient.request(
>       new SyndicationFeedAddAction({syndicationFeed})
> ```

consider checking that the provided feed doesn't have `id` (indicating it is an existing feed)

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html, line 10 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A5ALVELSv9De70Oy5:-L4A5ALVELSv9De70Oy6:bmf7vm0) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html#L10)):*
> ```HTML
>       {{'applications.content.syndication.details.header.cancel' | translate}}
>     </a>
>     <button *ngIf="_mode === 'edit'" class="kDeletehBtn kButtonDanger" pButton icon="kIcontrash"
> ```

is this button enabled while loading data or having handling errors on purpose? because it is outside of the areablocker

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html, line 16 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4sEi-e4nBq_1o34P:-L4A4sEi-e4nBq_1o34Q:b-cx55u) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html#L16)):*
> ```HTML
>     <button pButton class="kSaveBtn kButtonBranded"
>             (click)="_save()"
>             [disabled]="!_form.valid || !_currentDestinationFormState.isValid || (!_form.dirty && !_currentDestinationFormState.isDirty)"
> ```

OK, consider next time not creating a complex binding expression.  you can leave it here for this implementation

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html, line 153 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A5X8c51T_efSxjrAT:-L4A5X8c51T_efSxjrAU:b26dnxj) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html#L153)):*
> ```HTML
>               [targetOffset]="{'x': 197, 'y': _form.get('contentType').value === 'playlist' ? 10 : -38}"
>               [closeBtn]="false">
>   <div class="kWarningMoreInfoTooltip">
> ```

is this not wrapped with `ng-template` by design?

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 79 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A6H3Tc4rB0WHvvsCQ:-L4A6H3Tc4rB0WHvvsCR:biixorl) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L79)):*
> ```XML
> 
>   private _fillAvailablePlaylists(): void {
>     this._availablePlaylists = this.playlists.map(playlist => ({value: playlist, label: playlist.name || playlist.id}));
> ```

is it fair assumption that when this component is created the playlists were already fetched and provided by the input binding?

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 123 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A6WxnAoosib05mzpT:-L4A6WxnAoosib05mzpU:bki1pbn) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L123)):*
> ```XML
>       .subscribe(response => {
>         this._isBusy = false;
>         if (!this._isReady) {
> ```

this condition should wrap the whole content of this method including the `_queryData`

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 135 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A6lKlUQBnc8XaJg3-:-L4A6lKlUQBnc8XaJg30:b-5h3rqx) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L135)):*
> ```XML
>               warningCount: showEntriesCountWarning ? response.entriesCount.requireTranscodingCount : null,
>               flavorName: (() => {
>                 if (!showEntriesCountWarning) {
> ```

consider moving this method to a const variable above this expression

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 174 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A7-__NoaPqsm41U2X:-L4A7-__NoaPqsm41U2Y:b-s8x9f4) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L174)):*
> ```XML
>   private _queryData(): Observable<{ players: KalturaUiConf[], flavors: KalturaFlavorParams[], entriesCount?: KalturaSyndicationFeedEntryCount }> {
>     if (this._mode === 'edit' && (!this.feed || !this.feed.id)) {
>       return Observable.throw('Unable to load additional feed data');
> ```

please provide a more concrete error message

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 208 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A7VXqPZX7gEMDaXPq:-L4A7VXqPZX7gEMDaXPr:b4mzzzp) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L208)):*
> ```XML
> 
>   private _restartFormData(): void {
>     this._form.reset({
> ```

you reset the form in `onInit` but check for feed nullable mode during prepare. this will break the app. 

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 229 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A7yZH2aJLrEJMdNE7:-L4A7yZH2aJLrEJMdNE8:b-hu1up6) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L229)):*
> ```XML
>       !this._currentDestinationFormState.isValid ||
>       (!this._form.dirty && !this._currentDestinationFormState.isDirty)) {
>       return undefined;
> ```

I assume you should not reach this point? consider logging it

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 233 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ACdbsyZv2cxB-_aLd:-L4ACdbsyZv2cxB-_aLe:b2z4sr4) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L233)):*
> ```XML
> 
>     const syndicationFeed = this.destinationComponent.getData();
>     syndicationFeed.name = this._form.get('name').value;
> ```

I saw in the `.getData()` that you might return a null value so you must address it here as-well

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 261 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A8OrMOX8hcVzpgsGV:-L4A8OrMOX8hcVzpgsGW:btq4xk1) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L261)):*
> ```XML
>           buttons: [
>             {
>               label: this._appLocalization.get('app.common.retry'),
> ```

I'm not sure we should allow retry unless this is network issue. lets discuss

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 292 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A8dbDUevLPRKT0v2G:-L4A8dbDUevLPRKT0v2H:b-2guby3) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L292)):*
> ```XML
>           buttons: [
>             {
>               label: this._appLocalization.get('app.common.retry'),
> ```

same here

---

*[src/applications/content-syndication-app/feeds/feeds-list/feeds-list.component.ts, line 55 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AEw4EbaiaiRoNin4q:-L4AEw4EbaiaiRoNin4r:b-lrp1b4) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-list/feeds-list.component.ts#L55)):*
> ```XML
>   ngAfterViewInit() {
> 
>     // this.addNewFeed.state$
> ```

cleanup?

---

*[src/applications/content-syndication-app/feeds/feeds-list/feeds-list.component.ts, line 214 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AFKZ8UUMnGkfh65Zq:-L4AFKZ8UUMnGkfh65Zr:b-ghtim2) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-list/feeds-list.component.ts#L214)):*
> ```XML
>     // first the list of playlists and afterwards the feed list
>     // (while blocking the list to prevent the user from adding/editing feed without playlists provided)
>     this._feedsService.getPlaylists()
> ```

missing `_isReady` flag to prevent preparing multiple times

---

*[src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.html, line 1 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AFWGKhZP9zGFQ7mMz:-L4AFWGKhZP9zGFQ7mN-:b-mfzkfr) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.html#L1)):*
> ```HTML
> <k-area-blocker [showLoader]="(_feedsService.feeds.state$ | async)?.loading || this._deferredLoading"
> ```

lately we moved the loader/error handling to the container of the table, you should refactor this code as well

---

*[src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.ts, line 77 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AFh3RrE7d8qtOtPpD:-L4AFh3RrE7d8qtOtPpE:b-ooqkec) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.ts#L77)):*
> ```XML
> 
>   constructor(private _appLocalization: AppLocalization,
>               private _feedsService: FeedsService,
> ```

this will be removed once you will move the loader/error handling to the container of this component

---

*[src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.ts, line 175 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AFmYREnB_3pYIKLAv:-L4AFmYREnB_3pYIKLAw:b-lrp1b4) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.ts#L175)):*
> ```XML
>         }
>       },
>       // {
> ```

cleanup?

---

*[src/shared/kmc-shared/players/players-store.service.ts, line 22 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AHBoK_GQqQ6SIIhs6:-L4AHBoK_GQqQ6SIIhs7:b-i6kbhz) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/shared/kmc-shared/players/players-store.service.ts#L22)):*
> ```XML
>   public get(): Observable<{ items: KalturaUiConf[] }> {
>     let currentPageIndex = 0;
>     return Observable.create(observer => {
> ```

almost perfect. 

- use the same technique for the caching part [link](https://github.com/kaltura/kmc-ng/blob/2c696e99c6a92030f4bd923f6db919b8015c1e72/src/shared/content-shared/categories/categories-search.service.ts#L172)
- provide filter for entries/playlists - something like [this](https://github.com/kaltura/kmc-ng/blob/master/src/shared/kmc-shared/custom-metadata/metadata-profile-store.service.ts)
- use the same response profile, tags etc as [here](https://github.com/kaltura/kmc-ng/blob/d127ed923045664ab3f53e1a1ac80478859136a5/src/applications/preview-and-embed/preview-and-embed.service.ts#L21

---

*[src/shared/kmc-shared/players/players-store.service.ts, line 68 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AGu2oJm5ZAzD70gqU:-L4AGu2oJm5ZAzD70gqV:b-a26ted) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/shared/kmc-shared/players/players-store.service.ts#L68)):*
> ```XML
>     const pager = new KalturaFilterPager({pageSize, pageIndex});
> 
>     return <any>this._kalturaServerClient.request(new UiConfListAction({filter, pager}));
> ```

for what reason do you use here explicit casting?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: 36 of 51 files reviewed at latest revision, 37 unresolved discussions.

---

*[src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.html, line 19 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A0-0W9u0xQWPep_4j:-L4AbzoV69dm8gV6x5ts:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.html#L19)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

 'notSupported' should have a special handling. probably the same as 'idle' but disabled with the special tooltip for that state
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.ts, line 26 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A0nRX_b5ux2fZ5Fcx:-L4AemR3u3UdROyZHtu1:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.ts#L26)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

not having a text doesn't equal to feature not supported. should show 'idle' icon in disable mode 
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.ts, line 27 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A14d9dPQy1DceakVP:-L4AenQEDgkIVP33gTVV:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.ts#L27)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

I think we can remove this console warning it might confuse debugging process
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 93 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A2V3qldAGT7kqKOn5:-L4B-zSwUClfotVZOQ1x:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L93)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

in latest development we moved that part into _createDefaultFilterValue method
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 204 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A2v5NMYAozYYJGtAc:-L4B33KgPjwqGazyWoUn:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L204)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

if you map some items to undefined you need to also filter them out by adding after the `.map` operation `.filter(Boolean)` - see http://www.devign.me/javascript-tip-remove-falsy-items-out-of-an-array

here I would add a _logger.log message indicating that feed with 'id' was removed from list since it is not....
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 220 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A3bTl-eNA3VrXcft6:-L4B3Iza71boxYtn_2to:b-s7hiyy) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L220)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

is it ok querying for 500 items? was it taken from the legacy?
</blockquote></details>

yes it is taken from the legacy

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 284 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A46Fvu_l0N1SBGtUD:-L4B4Nci99Bwq2gt6sXU:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L284)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

since you only want to know if we have an error, `.find()` is more performant 
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 288 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4GVlvYEpaTDH5xgO:-L4B4i9r7qSltAjAfTpc:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L288)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

please provide more concrete error message
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 296 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4IyDzvT-VwPPXABu:-L4B4jOzM4xEJ5rjLxe8:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L296)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

cleanup?
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 392 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4W0hEVn5Xbj-UP4R:-L4B5h332NuP743Mou10:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L392)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

doesn't `syndicationFeed` include its' `id` value as a property? if so simplify the API
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds.service.ts, line 401 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4bYG3dizYlKLUn3S:-L4B6IeBEaU-ETTdpjQd:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds.service.ts#L401)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

consider checking that the provided feed doesn't have `id` (indicating it is an existing feed)
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html, line 10 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A5ALVELSv9De70Oy5:-L4BAeIpWH5Uh6qZAw4V:bx1wzz7) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html#L10)):*
> cationFeed = this.destinationComponent.getData();
need to put in disabled while loading

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html, line 16 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A4sEi-e4nBq_1o34P:-L4BAzR36KmOnzH1SZpB:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html#L16)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

OK, consider next time not creating a complex binding expression.  you can leave it here for this implementation
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html, line 153 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A5X8c51T_efSxjrAT:-L4BB4-dU0TmHLICofdl:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.html#L153)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

is this not wrapped with `ng-template` by design?
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 79 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A6H3Tc4rB0WHvvsCQ:-L4B7NXtLGsOLRLv-_4B:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L79)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

is it fair assumption that when this component is created the playlists were already fetched and provided by the input binding?
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 123 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A6WxnAoosib05mzpT:-L4B8AqnIkZuAvuBEETF:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L123)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

this condition should wrap the whole content of this method including the `_queryData`
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 135 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A6lKlUQBnc8XaJg3-:-L4B8iJ3cK0uu7WSIu5t:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L135)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

consider moving this method to a const variable above this expression
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 174 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A7-__NoaPqsm41U2X:-L4B8uT631gC1lKfuNtC:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L174)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

please provide a more concrete error message
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 208 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A7VXqPZX7gEMDaXPq:-L4BBnJLawVJQo_Po37a:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L208)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you reset the form in `onInit` but check for feed nullable mode during prepare. this will break the app. 
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 229 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A7yZH2aJLrEJMdNE7:-L4BCOGqHSGnMUWNVU6D:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L229)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

I assume you should not reach this point? consider logging it
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 233 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ACdbsyZv2cxB-_aLd:-L4BCraTYLKj3zQi5z0l:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L233)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

I saw in the `.getData()` that you might return a null value so you must address it here as-well
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 261 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A8OrMOX8hcVzpgsGV:-L4BHN3Gy6vceg2UbD5u:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L261)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

I'm not sure we should allow retry unless this is network issue. lets discuss
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts, line 292 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A8dbDUevLPRKT0v2G:-L4BHM_TozfosIoWP0W_:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feed-details/feed-details.component.ts#L292)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds-list/feeds-list.component.ts, line 55 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AEw4EbaiaiRoNin4q:-L4BDBckt5f0H_ThyICg:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-list/feeds-list.component.ts#L55)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

cleanup?
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds-list/feeds-list.component.ts, line 214 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AFKZ8UUMnGkfh65Zq:-L4BETL6iI2CKNCJ2PV0:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-list/feeds-list.component.ts#L214)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

missing `_isReady` flag to prevent preparing multiple times
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.html, line 1 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AFWGKhZP9zGFQ7mMz:-L4BPxWjRBBjOjLSep7F:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.html#L1)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

lately we moved the loader/error handling to the container of the table, you should refactor this code as well
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.ts, line 77 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AFh3RrE7d8qtOtPpD:-L4BOig6DQgUg61rzueL:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.ts#L77)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

this will be removed once you will move the loader/error handling to the container of this component
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.ts, line 175 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AFmYREnB_3pYIKLAv:-L4BHexnZzhwwniey6hD:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/feeds/feeds-table/feeds-table.component.ts#L175)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

cleanup?
</blockquote></details>

Done.

---

*[src/shared/kmc-shared/players/players-store.service.ts, line 68 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AGu2oJm5ZAzD70gqU:-L4BHq3i-deJL7dor_GX:bgstn5f) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/shared/kmc-shared/players/players-store.service.ts#L68)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

for what reason do you use here explicit casting?
</blockquote></details>

i copied it from the flavours-store and missed this part

---

*[src/shared/kmc-shared/players/players-store.service.ts, line 73 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4BtbHak7rHwJIwuuAg:-L4BtbHak7rHwJIwuuAh:bc57r33) ([raw file](https://github.com/kaltura/kmc-ng/blob/d8de4590dd1492a8864ef8515a53ca292efaf91f/src/shared/kmc-shared/players/players-store.service.ts#L73)):*
> ```XML
>           loadPlayers();
> 
>           return () => {
> ```

@esakal We need to unsubscribe and clear the cache when all subscriptions completed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: 36 of 51 files reviewed at latest revision, 36 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ABdrj9-pWfat_2oM7:-L4BtnU-HYHWOT6jligT:b-896fix) (no related file):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

since you did it nicely in players and also need to implement caching mechanism there - can you do it here as well?
</blockquote></details>

Done.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AE6i0v6UsgxN28usM:-L4BtpcegyzOJNXARUP7:b-896fix) (no related file):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

should add local search for players and flavors
</blockquote></details>

Done.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AEGqUBuNWyBiyU9Bf:-L4Btsbcjk1j_eClkr03:bwv8x0e) (no related file):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

when creating a new feed and selecting destination there is a flicker of the validation on the destination component
</blockquote></details>

Amir is aware of that, can

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ATLuochlsHXMx6YV_:-L4F_T7qVC_46b9xz2Rk:b-896fix) (no related file):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

should clear cache of players when added/deleted player in 'studio'

Use the event provided by Amir
https://github.com/kaltura/kmc-ng/blob/master/src/shared/content-shared/categories/categories-search.service.ts#L55
</blockquote></details>

Done.

---

*[src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.html, line 1 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4A-cEGe7frDvKMcVSs:-L4F_WRLy7IsOGHC8GV7:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/applications/content-syndication-app/copy-to-clipboard/copy-to-clipboard.component.html#L1)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

this component should be in mc-shared
</blockquote></details>

Done.

---

*[src/shared/kmc-shared/players/players-store.service.ts, line 22 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AHBoK_GQqQ6SIIhs6:-L4F_Z3mdv7PYuNxfzy9:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/9713acb98d7bab6a0ff79b5992eb0ea0469ab77c/src/shared/kmc-shared/players/players-store.service.ts#L22)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

almost perfect. 

- use the same technique for the caching part [link](https://github.com/kaltura/kmc-ng/blob/2c696e99c6a92030f4bd923f6db919b8015c1e72/src/shared/content-shared/categories/categories-search.service.ts#L172)
- provide filter for entries/playlists - something like [this](https://github.com/kaltura/kmc-ng/blob/master/src/shared/kmc-shared/custom-metadata/metadata-profile-store.service.ts)
- use the same response profile, tags etc as [here](https://github.com/kaltura/kmc-ng/blob/d127ed923045664ab3f53e1a1ac80478859136a5/src/applications/preview-and-embed/preview-and-embed.service.ts#L21
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 3 of 3 files at r2, 10 of 12 files at r3, 11 of 11 files at r4.
Review status: 46 of 55 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4AEGqUBuNWyBiyU9Bf:-L4k_idSp6jeiZU7OJjn:b-87tse6) (no related file):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

Amir is aware of that, can
</blockquote></details>

+@amirch1 this is something the will became a bug. it there a way to bypass it?

---

*[src/shared/kmc-shared/flavours/flavours-store.service.ts, line 27 at r4](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4kapVtTBvRzG6dak9h:-L4kXSJj3HuphbKwNpdf:b-kra8a8) ([raw file](https://github.com/kaltura/kmc-ng/blob/ba2fc431cb25ed806116715dd7e704f3c36738ce/src/shared/kmc-shared/flavours/flavours-store.service.ts#L27)):*
> ```XML
>       this._getFlavorsFilters$ = this._buildGetRequest().map(
>         response => {
>           return ({items: response ? response.objects : []});
> ```

you are missing the `.catch()` action that should clear the `_getFlavorsFilters` otherwise an error will be cached and replay to the users.

I added this implementation please test at runtime

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

esakal commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 46 of 55 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-:-L4kbiRmFh1dalvJDAAl:bnfp4nl)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 9 of 12 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

amirch1 commented:
Reviewed 29 of 49 files at r1, 3 of 3 files at r2, 8 of 12 files at r3, 6 of 11 files at r4, 9 of 12 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4oeQG_2fvoSGuHTxF0:-L4oeQGammZw_RA8j7Zw:b9hj748) (no related file):*
+@benstov - after bulk delete - the list is reloaded but the bulk selection interface remains even though no feed is selected

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

amirch1 commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-L4ofeDKLrBDpQaNT8Ha:-L4ofeDKLrBDpQaNT8Hb:b73nq20) (no related file):*
during sticky scroll of feeds list - missing grey separator:
![✂-1](https://files.reviewableusercontent.io/images/reviewable/3839950/4BKl95PRhkaZJeaIccXG9pI8.png)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 7 of 8 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[a discussion](https://beta.reviewable.io/reviews/kaltura/kmc-ng/363#-L4oeQG_2fvoSGuHTxF0:-L4p9Pps5_ZenasOWEc8:b-896fix) (no related file):*
<details><summary><i>Previously, amirch1 (Amir Chervinsky) wrote…</i></summary><blockquote>

+@benstov - after bulk delete - the list is reloaded but the bulk selection interface remains even though no feed is selected
</blockquote></details>

Done.

---

*[a discussion](https://beta.reviewable.io/reviews/kaltura/kmc-ng/363#-L4ofeDKLrBDpQaNT8Ha:-L4pAXfE4PbkQDtfOyAp:b-896fix) (no related file):*
<details><summary><i>Previously, amirch1 (Amir Chervinsky) wrote…</i></summary><blockquote>

during sticky scroll of feeds list - missing grey separator:
![✂-1](https://files.reviewableusercontent.io/images/reviewable/3839950/4BKl95PRhkaZJeaIccXG9pI8.png)

</blockquote></details>

Done.

---

*[src/shared/kmc-shared/flavours/flavours-store.service.ts, line 27 at r4](https://beta.reviewable.io/reviews/kaltura/kmc-ng/363#-L4kapVtTBvRzG6dak9h:-L4pA_h91rFfYN55yvOe:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/ba2fc431cb25ed806116715dd7e704f3c36738ce/src/shared/kmc-shared/flavours/flavours-store.service.ts#L27)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you are missing the `.catch()` action that should clear the `_getFlavorsFilters` otherwise an error will be cached and replay to the users.

I added this implementation please test at runtime
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->

amirch1 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 54 of 56 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363#-:-L4pL1zPMRlMG5KbIyJ8:bnfp4nl)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 2 of 2 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/363)*
<!-- Sent from Reviewable.io -->
